apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Precondition.cs

```diff
@@ -65,5 +66,19 @@ public sealed class Precondition
 
         // Only used by tests
         internal static Precondition FromProto(V1Beta1.Precondition proto) => proto == null ? null : new Precondition(proto);
+
+        /// <summary>
+        /// Compares this precondition with another for equality.
+        /// </summary>
+        /// <param name="other">The precondition to compare this one with</param>
+        /// <returns><c>true</c> if this precondition is equal to <paramref name="other"/>; <c>false</c> otherwise.</returns>
+        public bool Equals(Precondition other) => other != null && Equals(Proto, other.Proto);
+
+        /// <inheritdoc />
+        public override bool Equals(object obj) => Equals(obj as Precondition);
+
+        // Value for null is arbitrary, but not 0 as that's what a MustExist gives too.
```

Comment:
nit: Can this comment be explained better? I don't understand what this means.

Reply:
Sure, will do. It's explaining the "?? 5172" bit, but I can make it clearer.

Reply:
I got the `?? 5172` bit; the "that's what MustExist gives too" bit confused me.

Reply:
Ah, I see. It so happens that MustExist.Proto.GetHashCode() returns 0, so it's best if None (no proto) returns something other than 0.